core/controllers/question_editor.py

```diff
@@ -47,11 +36,11 @@ def post(self):
 
         question_dict = self.payload.get('question_dict')
         if (
-                ('id' not in question_dict) or
                 ('question_state_data' not in question_dict) or
                 ('language_code' not in question_dict)):
             raise self.InvalidInputException
 
+        question_dict['id'] = question_services.get_new_question_id()
```

Comment:
Validate that the incoming question_dict does not contain an ID or a version yet (or that the version is 1).

I don't think you should overwrite question_state_schema_version -- rather, compare it to the feconf constant and reject if there's a mismatch. It's possible that the version changed while the user was editing the question (if we did a server push at that time).

Reply:
Done.